bin/ajax

```diff
@@ -284,7 +287,43 @@ def clean_unregistered(delete_live=False):
     log.info(f'clean_unregistered::\tfound {len(run_ids)} runs stored on'
              f'{folder_to_check}. Checking that each is in the runs-database')
     for run_id in run_ids:
-        remove_if_unregistered(int(run_id), delete_live=delete_live)
+        run_number = int(run_id)
+        remove_if_unregistered(run_number, delete_live=delete_live)
+
+
+def clean_old_hash():
+    """
+    Query the database for data that was processed with other settings of straxen that
+    lead to a different linage hash. If we can find something on this host that is not in
+    correspondence with the latest hash: delete it.
+    """
+    rds = run_coll.find({'data.type.$': {'$nin': low_level_data}, 'data.host': hostname})

```

Comment:
If we expect this to run more than once, we'll want to index this field, and I think that this field would be a messy and inefficient index. If we made `data` its own collection, this would be rather simpler.

I'd almost suggest doing this via the filesystem rather than the database.

Reply:
This would only be done once each hour although we can up this time as most of the time you wouldn't really expect a lot of old data lying around. 

I'm not sure if doing it via the filesystem would be of much help given that we in the end want to sync the rdb and the filesystem.

Reply:
That query is going to run increasingly slowly as more and more runs accumulate, while the filesystem will generally have a "constant" number of items (certainly less than the number of runs). It will probably be faster to query the disk, perform whatever changes are necessary, and then update the database, rather than the other way around. If we reduce the frequency of this (1/day?) it'll be less of an issue.

Reply:
Hmm okay, actually doing it over the fs is easier. Thanks for the suggestion!